src/dimension.h

```diff
@@ -39,6 +39,7 @@ typedef struct Dimension
 #define IS_CLOSED_DIMENSION(d) ((d)->type == DIMENSION_TYPE_CLOSED)
 #define IS_VALID_OPEN_DIM_TYPE(type)                                                               \
 	(IS_INTEGER_TYPE(type) || IS_TIMESTAMP_TYPE(type) || ts_type_is_int8_binary_compatible(type))
+#define IS_CORRELATED_CONSTRAINT(d) ((d)->fd.correlated == true)
```

Comment:
Maybe it makes sense to expand DimensionType to include DIMENSION_TYPE_CORRELATED instead of adding a boolean?

Reply:
Ok, done.